internal/plugin/tftypes.go

```diff
@@ -1181,3 +1181,64 @@ func (b *tfObjectSlice) String() string {
 		return fmt.Sprintf("%s", b.Val)
 	}
 }
+
+type dynamicPseudoTypeBlock struct {
+	Object        *tfObject
+	OriginalValue tftypes.Value
+}
+
+func newDynamicPseudoTypeBlock() *dynamicPseudoTypeBlock {
+	return &dynamicPseudoTypeBlock{Object: newTfObject()}
+}
+
+// FromTFValue returns unmarshals a tftypes.Value into itself.
+func (d *dynamicPseudoTypeBlock) FromTFValue(value tftypes.Value) error {
+	if !value.IsKnown() {
+		// RetryJoin are a DynamicPseudoType but the value is unknown. Terraform expects us to be a
```

Comment:
`RetryJoin` is cruft from the old implementation that was tied to the storage config type.